pex/tools/commands/venv.py

```diff
@@ -24,18 +27,30 @@
 from pex.venv_bin_path import BinPath
 
 if TYPE_CHECKING:
+    import typing
     from typing import Iterable, Iterator, Optional, Tuple
 
 logger = logging.getLogger(__name__)
 
 
+def _relative_symlink(
+    src,  # type: str
+    dst,  # type: str
+):
+    # type: (...) -> None
+    dst_parent = os.path.dirname(dst)
+    rel_src = os.path.relpath(src, dst_parent)
+    os.symlink(rel_src, dst)
+
+
 # N.B.: We can't use shutil.copytree since we copy from multiple source locations to the same site
 # packages directory destination. Since we're forced to stray from the stdlib here, support for
 # hardlinks is added to provide a measurable speed up and disk space savings when possible.
 def _copytree(
     src,  # type: str
     dst,  # type: str
     exclude=(),  # type: Tuple[str, ...]
+    symlink=False,  # type: bool
 ):
```

Comment:
Why copy the internals of the `src` and `dst` rather than directly creating `dst` as a symlink? In cases where we are creating symlinks, it's not like we can delete `src` afterward, so `src` must be long lived.

...if this is explained below in the "In the symlink case, in order to share all generated" comment, then I'm not quite understanding it.

Reply:
Its not explained, but backing up to the locally assumed domain specific knowledge of a venv and site-packages, `dst` is `site-packages` and `src` is some installed wheel chroot. Those typically look like:
```
ls -l ~/.pex/installed_wheels/23de6e1192493377da687afdbb2a2cf11e6d613a/Django-3.2.10-py3-none-any.whl/
total 12
drwxr-xr-x  2 jsirois jsirois 4096 Dec 13 10:26 bin
drwxr-xr-x 18 jsirois jsirois 4096 Dec 13 10:25 django
drwxr-xr-x  2 jsirois jsirois 4096 Dec 13 10:26 Django-3.2.10.dist-info
```
So when you symlink this into `site-packages` you need:
```
site-packages/
  django -> ...
  Django-3.2.10.dist-info -> ...
  foo_dist -> ...
  foo-dist-1.0.0.dist-info -> ...
```
  
So, basically, each `src` has more than one item that must be linked into one big flat `dst`. There are many `src`s.

Reply:
Got it. Thanks!